Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -15,21 +15,64 @@ vector<uint> dockingModuleEquipmentIds;
 // Above how much cargo capacity, should a ship be rejected as a docking user?
 int cargoCapacityLimit = 275;
 
+// How much time will player be given before kick if carrier wants to jettison him.
+int jettisonKickTime = 15;
+
 uint connSystemID = CreateID("li06");
 
-///////////////////////////////////////////////////////////////////////////////////////////////////////////////
-// Function handling changing systems when undocking from a carrier
-void JumpToLocation(uint client, uint system, Vector pos, Matrix ornt)
+// Delayed actions, which need to be done. Look at HkTimerCheckKick().
+vector<ActionJettison> jettisonList;
+
+void HkTimerCheckKick()
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	for (vector<ActionJettison>::iterator it = jettisonList.begin(); it != jettisonList.end();)
+	{
+		it->timeLeft--;
+		if (it->timeLeft == 0)
+		{
+			uint checkCarrierClientID = HkGetClientIdFromCharname(it->carrierCharname);
+			uint checkDockedClientID = HkGetClientIdFromCharname(it->dockedCharname);
+
+			// If both carrier and docked ship are still at server.
+			if (checkDockedClientID != -1 && checkCarrierClientID != -1)
+			{
+				// If the client is still docked
+				if (mobiledockClients[checkDockedClientID].mobileDocked)
+				{
+					HkKickReason(ARG_CLIENTID(checkDockedClientID), L"Forced undocking.");
+
+					// If wscDockedWithCharname, that does mean that carrier died and next message is not required.
+					if(!mobiledockClients[checkDockedClientID].wscDockedWithCharname.empty())
+					PrintUserCmdText(checkCarrierClientID, L"Jettisoning ship...");
+				}
+			}
+
+			jettisonList.erase(it);
+		}
+		else
+		{
+			it++;

```

Comment:
It really makes if one of elements in vector was erased DURING iterating through it in for loop.

Reply:
Did you test this change? It seems incredibly dangerous.

Reply:
I did tests before. Server crashed until I removed automatic ++ of iterator and made it to be done only if no elements were erased from vector.

Reply:
What if we start from the end and go through the list backwards with --, removing items along the way? This code appears wrong as-is.

Reply:
The code works well, but there seems to be better alternative. Should test that later.

Reply:
Instead of discarding the result of `jettisonList.erase(it);`, use it like this:
`it = jettisonList.erase(it);`
https://en.cppreference.com/w/cpp/container/vector/erase
Then this will all make sense I think.

Reply:
Done.